frontend/src/components/maps/leaflet/Map.tsx

```diff
@@ -237,11 +237,7 @@ const Map: React.FC<MapProps> = ({
   const handleMapFilterChange = async (filter: IPropertyFilter) => {
     const compareValues = (objValue: any, othValue: any) => {
       return whitelistedFilterKeys.reduce((acc, key) => {
-        return (
-          (isEqual(objValue[key], othValue[key]) ||
-            (isEmpty(objValue[key]) && isEmpty(othValue[key]))) &&
-          acc
-        );
+        return (isEqual(objValue[key], othValue[key]) || (!objValue[key] && !othValue[key])) && acc;
```

Comment:
Just replaced `isEmpty(obj[key])` with `!obj[key]` which handles the case of the value being true correctly